### PR TITLE
[HackDay][WebProfilerBundle]Display the main template on the debug toolbar

### PR DESCRIPTION
**Pull request by vincentmoulene**
| Q             | A
| ------------- | ---
| Branch?       | master
| Bug fix?      | no
| New feature?  | yes
| BC breaks?    | no
| Deprecations? | no
| Tests pass?   | yes
| Fixed tickets | #
| License       | MIT
| Doc PR        | None

It's my first PR for the Symfony community. 
I thought it would be useful to display the main template in the debug toolbar.
Eg : 
capture d ecran 2018-12-08 a 16 16 48

Thanks for all and thanks for you feedbacks.

#SymfonyConHackday2018

### CODE REVIEW

**Review by ro0NL — COMMENTED** [review]
im a bit worried the logic is hard to reason about, by assuming that it's always one "main" render call (`return $this->render('main_template.html.twig')`). Whereas technically we just show the first render call and call it the main one :} im not sure that's OK.

File: src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/request.html.twig [hunk 1]
```diff
@@ -55,6 +55,12 @@
                 <span>{{ collector.route|default('n/a') }}</span>
             </div>
 
+            {%- set keystemplates = profile.collectors['twig'].getTemplates|keys|default(false) -%}
```
Comment (ro0NL) [inline src/Symfony/Bundle/WebProfilerBundle/Resources/views/Collector/request.html.twig]: this creates a dependency with the twig collector which can [break](https://github.com/symfony/symfony/blob/7b03bff57a2782edc6558d57df7ccff99ed0f1a9/src/Symfony/Component/HttpKernel/Profiler/Profile.php#L242), to me that implies it belongs in the twig toolbar block at least.

**Comment by jvasseur** [comment]
I agree with @ro0NL here, the first call to render is not necessary the main template, one example is when you first render an template to send an email and then send a response.

**Comment by nicholasruunu** [comment]
Really good idea, would save me time. 👍

**Comment by ro0NL** [comment]
Also see a related past discussion on the topic: https://github.com/symfony/symfony/pull/24236#issuecomment-330048491 / https://github.com/symfony/symfony/pull/24236#pullrequestreview-64189250

And perhaps more ultimately: #28056

**Comment by fabpot** [comment]
It was not done before for the reasons explained by others: there is not a "main" template concept in Symfony. And the first one might not be the "main" one. I suggest to close as having a feature that does not work for all use cases is not something we want in core.

**Pull request closed without merging**